pom.xml

```diff
@@ -407,6 +405,16 @@
                 </repository>
             </distributionManagement>
         </profile>
+
+        <profile>
+            <id>jdk9+</id>
+            <activation>
+                <jdk>[9,)</jdk>
+            </activation>
+            <properties>
+                <maven.compiler.release>${java.version}</maven.compiler.release>
```

Comment:
When building by JDK 8, it would fail on this property. But when building by 9+, we need this for backward compatibility, see https://www.baeldung.com/java-compiler-release-option